### PR TITLE
Item usage

### PR DESCRIPTION
**Pull request by jamesshieh**
Adding the ability to use items in the menu and battle.

Now items can carry with them any skill which will impart effects on the target during battle and out of battle. 

Can also be used to make spell scrolls items and such!

Each item needs its own skill right now because the battle action still uses the skill.name to write out the effects of the item usage. 

Also inventory doesn't update after you use an item yet.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ module.exports = function(grunt) {
       }
     },
     watch: {
-      files: ['browser/js/**/*.js', 'lib/*.js'],
+      files: ['browser/js/**/*.js', 'lib/*.js', 'browser/js/**/**/*.js'],
```
Comment (amfeng) [inline Gruntfile.js]: Was it not working? ** means multiple levels.

File: lib/battle_action.js [hunk 2]
```diff
@@ -18,11 +19,68 @@ BattleAction.prototype.execute = function() {
     this.defend();
   } else if (this.action === "heal") {
     this.heal();
+  } else if (this.action === "regainMp") {
```
Comment (amfeng) [inline lib/battle_action.js]: What's the difference between heal and regainMp? (regain MP seems kind of random)

File: lib/battle_action.js [hunk 3]
```diff
@@ -18,11 +19,68 @@ BattleAction.prototype.execute = function() {
...
+  var that = this;
+  this.targets.forEach(function (t) {
+    var result = {};
+    result.targetName = t.name;
+    result.type = "heal";
+    if (that.participant.mp >= that.skill.mpCost) {
+
+
+      // Remove defending because you took an action
+      that.participant.defending = false;
```
Comment (amfeng) [inline lib/battle_action.js]: Could defending just last one turn? We could just have a thing that checks in the beginning of your turn and turns off things that shouldn't be on (e.g. defense, status effects)

File: lib/battle_action.js [hunk 4]
```diff
@@ -18,11 +19,68 @@ BattleAction.prototype.execute = function() {
...
+    result.targetName = t.name;
+    result.type = "item";
+    result.itemName = that.skill.name;
+
+    var amount = Math.floor(that.skill.baseDamage);
+    result.amount = amount;
+    if (that.skill.actionType === "heal") {
+      t.heal(amount);
+      result.attribute = "hp";
+    } else if (that.skill.actionType === "regainMp") {
```
Comment (amfeng) [inline lib/battle_action.js]: Shouldn't mp just be a property of heal? e.g. elixers heal both hp and mp

File: lib/battle_action.js [hunk 5]
```diff
@@ -18,11 +19,68 @@ BattleAction.prototype.execute = function() {
...
+      t.regainMp(regainAmount);
+      result.regain = regainAmount;
+      // Apply status effects if the attack has any
+      that.applyStatuses(t);
+    }
+    that.results.push(result);
+  });
+
+  // Pay MP cost
+  that.participant.mp -= that.skill.mpCost;
```
Comment (amfeng) [inline lib/battle_action.js]: this?

File: lib/battle_action.js [hunk 6]
```diff
@@ -18,11 +19,68 @@ BattleAction.prototype.execute = function() {
...
+  // Increase AP by skill cost
+  this.participant.ap += that.skill.apCost;
```
Comment (amfeng) [inline lib/battle_action.js]: that?

File: Gruntfile.js (hunk 1 above)
Comment (jamesshieh) [inline Gruntfile.js]: It wasn't refreshing on inventory changes

File: lib/battle_action.js (hunk 2 above)
Comment (jamesshieh) [inline lib/battle_action.js]: hmm I guess i could combine them into one and make everything that calls it push either "hp" or "mp"

havent cleaned that up yet

File: lib/battle_action.js (hunk 3 above)
Comment (jamesshieh) [inline lib/battle_action.js]: Hmm... could remove defending by default when a battle action is executed instead. will consolidate it into 1 place

File: lib/battle_action.js (hunk 4 above)
Comment (jamesshieh) [inline lib/battle_action.js]: Cant handle elixirs yet :P going to have to change the item schemas probably

**Pull request merged**